projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -282,6 +282,131 @@ describe('DatagridComponent', () => {
                 });
             });
 
+            describe('datagridSelection getter', () => {
+                describe('when selectionType is set to multi', () => {
+                    beforeEach(function (this: HasFinderAndGrid): void {
+                        this.hostComponent.selectionType = GridSelectionType.Multi;
+                        this.finder.detectChanges();
+                    });
+
+                    it('returns same reference when there is no selection', fakeAsync(function (
+                        this: HasFinderAndGrid
+                    ): void {
+                        const selection = this.hostComponent.grid.datagridSelection;
+                        tick();
+                        this.finder.detectChanges();
+                        expect(this.hostComponent.grid.datagridSelection).toEqual([]);
+                        expect(this.hostComponent.grid.datagridSelection).toBe(selection);
+                    }));
+                    it('returns same reference when there is a selection that was not changed', fakeAsync(function (
+                        this: HasFinderAndGrid
+                    ): void {
+                        this.clrGridWidget.getSelectionLabelForRow(0).click();
+                        tick();
+                        this.finder.detectChanges();
+                        const selection = this.hostComponent.grid.datagridSelection;
+                        tick();
+                        this.finder.detectChanges();
+                        expect(this.hostComponent.grid.datagridSelection).toBe(selection);
+                    }));
+                    it('returns new reference when there is a selection that was changed', fakeAsync(function (
+                        this: HasFinderAndGrid
+                    ): void {
+                        // Select an item
+                        this.clrGridWidget.getSelectionLabelForRow(0).click();
+                        tick();
+                        this.finder.detectChanges();
+                        const selection = this.hostComponent.grid.datagridSelection;
+                        // Deselect the item
+                        this.clrGridWidget.getSelectionLabelForRow(0).click();
+                        tick();
+                        this.finder.detectChanges();
+                        // Select the same item
+                        this.clrGridWidget.getSelectionLabelForRow(0).click();
+                        tick();
+                        this.finder.detectChanges();
+                        expect(this.hostComponent.grid.datagridSelection).toEqual(selection);
```

Comment:
Can we expect that this.hostComponent.grid.datagridSelection[0] is the same reference as selection[0]?

Reply:
yes, I have added that check also